builder/pipeline_test.cc

```diff
@@ -988,60 +1003,68 @@ TEST(pipeline, copied_result) {
 }
 
 TEST(pipeline, concatenated_result) {
-  // Make the pipeline
-  node_context ctx;
+  for (bool no_alias_buffers : {false, true}) {
+    // Make the pipeline
+    node_context ctx;
 
-  auto in1 = buffer_expr::make(ctx, "in1", sizeof(short), 2);
-  auto in2 = buffer_expr::make(ctx, "in2", sizeof(short), 2);
-  auto out = buffer_expr::make(ctx, "out", sizeof(short), 2);
+    auto in1 = buffer_expr::make(ctx, "in1", sizeof(short), 2);
+    auto in2 = buffer_expr::make(ctx, "in2", sizeof(short), 2);
+    auto out = buffer_expr::make(ctx, "out", sizeof(short), 2);
 
-  auto intm1 = buffer_expr::make(ctx, "intm1", sizeof(short), 2);
-  auto intm2 = buffer_expr::make(ctx, "intm2", sizeof(short), 2);
+    auto intm1 = buffer_expr::make(ctx, "intm1", sizeof(short), 2);
+    auto intm2 = buffer_expr::make(ctx, "intm2", sizeof(short), 2);
 
-  intm1->dim(1).bounds = in1->dim(1).bounds;
-  intm2->dim(1).bounds = in2->dim(1).bounds;
+    intm1->dim(1).bounds = in1->dim(1).bounds;
+    intm2->dim(1).bounds = in2->dim(1).bounds;
 
-  var x(ctx, "x");
-  var y(ctx, "y");
+    var x(ctx, "x");
+    var y(ctx, "y");
 
-  // In this pipeline, the result is copied to the output. We should just compute the result directly in the output.
-  func add1 = func::make(add_1<short>, {{{in1, {point(x), point(y)}}}}, {{{intm1, {x, y}}}});
-  func add2 = func::make(add_1<short>, {{{in2, {point(x), point(y)}}}}, {{{intm2, {x, y}}}});
-  func concatenated = func::make_copy(
-      {intm1, {point(x), point(y)}}, {intm2, {point(x), point(y - in1->dim(1).extent())}}, {out, {x, y}});
+    // In this pipeline, the result is copied to the output. We should just compute the result directly in the output.
+    func add1 = func::make(add_1<short>, {{{in1, {point(x), point(y)}}}}, {{{intm1, {x, y}}}});
+    func add2 = func::make(add_1<short>, {{{in2, {point(x), point(y)}}}}, {{{intm2, {x, y}}}});
+    func concatenated = func::make_copy(
+        {intm1, {point(x), point(y)}}, {intm2, {point(x), point(y - in1->dim(1).extent())}}, {out, {x, y}});
 
-  // TODO(https://github.com/dsharlet/slinky/issues/21): The checks on the input bounds are overzealous in this case. We
-  // shouldn't need to disable checks.
-  pipeline p = build_pipeline(ctx, {in1, in2}, {out}, build_options{.no_checks = true});
+    // TODO(https://github.com/dsharlet/slinky/issues/21): The checks on the input bounds are overzealous in this case.
+    // We shouldn't need to disable checks.
+    pipeline p =
+        build_pipeline(ctx, {in1, in2}, {out}, build_options{.no_checks = true, .no_alias_buffers = no_alias_buffers});
```

Comment:
FWIW, the `.member_init` syntax isn't legal C++17... it's a common compiler extension but wasn't codified for C++ until C++20 (we must not have all the 'disable extensions' settings on). Another vote for enabling C++20 here :-)

Reply:
I didn't realize that. We're already using this pattern elsewhere, and only in tests, I'll file an issue.